### PR TITLE
[Admin] Loosely couple with Shop in case of customer impersonation

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/8271 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/DependencyInjection/SyliusAdminExtension.php [hunk 1]
```diff
@@ -33,6 +33,13 @@ public function load(array $config, ContainerBuilder $container): void
 
         $container->setParameter('sylius.admin.notification.enabled', $config['notifications']['enabled']);
         $container->setParameter('sylius.admin.notification.frequency', $config['notifications']['frequency']);
+        $container->setParameter('sylius.admin.shop_enabled', false);
+
+        $bundles = $container->getParameter('kernel.bundles');
+        if (array_key_exists('SyliusShopBundle', $bundles)) {
+            $loader->load('services/shop.xml');
```
Comment (pamil) [inline src/Sylius/Bundle/AdminBundle/DependencyInjection/SyliusAdminExtension.php]: Let's follow the current pattern of having optional services in `integrations` directory, like `integrations/shop.xml` here.

File: src/Sylius/Bundle/AdminBundle/Twig/ImpersonateCustomerExtension.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Twig;
+
+final class ImpersonateCustomerExtension extends \Twig_Extension
```
Comment (pamil) [inline src/Sylius/Bundle/AdminBundle/Twig/ImpersonateCustomerExtension.php]: It's not related to impersonating customer, it's rather a `ShopExtension` or something like that.

File: src/Sylius/Bundle/AdminBundle/Twig/ImpersonateCustomerExtension.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * @param bool $isShopEnabled
+     */
+    public function __construct(bool $isShopEnabled)
+    {
+        $this->isShopEnabled = $isShopEnabled;
+    }
+
+    /**
+     * @return array|\Twig_Function
```
Comment (pamil) [inline src/Sylius/Bundle/AdminBundle/Twig/ImpersonateCustomerExtension.php]: `\Twig_Function` -> `\Twig_Function[]`

**Comment by pjedrzejewski** [comment]
Thank you Mateusz!

@lchrusciel Good idea for later! 👍

**Pull request merged**